### ISSUE TITLE
Overfitting Training Data, and What Neurons Were Used In Shapes?

### ISSUE THREAD

**Issue by windowshopr** [autonomio/talos#476]
#### 1) Confirm the below

I have searched the user guide, and other issues and haven't found an answer for these two questions:

1. I would like to know why in my final output .csv file, the training data is near 100% accuracy, yet the testing accuracy is around 45% for a binary classification problem, using binary_crossentropy as the metric and doing a 60/40 train/test split. There seems to be quite a bit of overfitting going on. Wondering how I can combat that because I thought by default the `clear_tf_session ` was True by default. Wondering what might be going on here. I do have early_stopping callback set to moderate which I thought might help?

2. When using the shapes arguments, is there a way to potentially add what the subsequent neuron widths were for each layer to the output.csv file? For example, if I have the following parameters:

```
p = {'first_neuron':[32, 64, 128, 256],
     'hidden_layers':[1, 2, 3],
     'batch_size': [32, 64, 128, 256],
     'epochs': [2000],
     'dropout': [0, 0.25, 0.5],
     'shapes':['brick', 'funnel', 'triangle'],
     'optimizer': ['Nadam', 'Adam'],
     'losses': ['binary_crossentropy'],
     'activation':['relu', 'elu'],
     'last_activation': ['sigmoid']}
```

...if it was currently scanning the 3 layer, triangle shape, could it possibly add 2 new columns to the .csv file showing how many 'first_neuron's were used in the subsequent layers? Also I'm just a little unclear on how the shapes work. Am I correct in assuming that brick means it'll use the same number of neurons for each layer, funnel will decrease the neuron width for the following layer(s), and triangle would do the opposite?

Love the tool, first time running it tonight and if I can get these two things figured out, I'll be one happy camper! Thanks a lot!

**Comment by windowshopr** [comment]
...Also in the user guide, there needs to be a change made in the Scan() documentation. Currently it reads:


`clear_tf_session | True | Clear tensorflow session after each round`

...but I see [here](https://github.com/autonomio/talos/issues/206) that this has been changed to:

`clear_session='tf', # or 'theano', and 'cntk'`

Just noticed that. Thanks!

**Comment by mikkokotila** [comment]
It is because your model is severely overtraining. This has nothing to do with Talos, but is related with your model configuration. Try adding more data, using regulizers, and other things that are generally useful for overfitting problem.

Also are you sure you want to use 2000 epochs?

**Comment by windowshopr** [comment]
@mikkokotila I will try adding weight regularization and learning rates to the param dictionary and see where that gets me. I have it at 2000 epochs but with early stopping set to moderate, 200 epochs would be enough time to stop looking for a better model as that's what I have set in my standalone Tensorflow model, however in that I am using a for loop and Cross Validation to get a general overview of how well that model architecture performs on the data.

I guess that's another question, if Talos uses CV during training or just during the evaluate step? Thanks again!
